File: articles/calculus-for-systems-modeling/product-rule-interaction-effects/fortran/src/product_rule.f90
program product_rule_demo
  implicit none
  real :: a, b, da, db, ca, cb
  a = 120.0
  b = 1.5
  da = 4.0
  db = 0.03
  ca = da * b
  cb = a * db
  print *, "contribution_from_a=", ca
  print *, "contribution_from_b=", cb
  print *, "total_derivative=", ca + cb
end program product_rule_demo

File: articles/calculus-for-systems-modeling/product-rule-interaction-effects/c/src/product_rule.c
#include <stdio.h>

int main(void) {
    double a = 120.0, b = 1.5, da = 4.0, db = 0.03;
    double ca = da * b;
    double cb = a * db;
    printf("contribution_from_a=%f\n", ca);
    printf("contribution_from_b=%f\n", cb);
    printf("total_derivative=%f\n", ca + cb);
    return 0;
}
